src/main/java/Bot/DateConverter.java

```diff
@@ -0,0 +1,115 @@
+package Bot;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ Responsible for providing methods for formatting user input date string into readable date string format
+ */
+public class DateConverter {
+
+    /**
+     * Converts the given date string to a readable format. It determines if the
+     * input is in a datetime format or a date-only format and applies the appropriate conversion.
+     *
+     * @param input The date string to be converted.
+     * @return The formatted date string or an error message if the format is invalid.
+     */
+    public static String convertDate(String input) {
+        if (isDateTimeFormat(input)) {
+            return convertDateTime(input);
+        } else if (isDateFormat(input)) {
+            return formatDate(input);
+        } else {
+            return "Invalid date format!";
+        }
+    }
+
+    /**
+     * Checks if the input string is in the "d/M/yyyy HHmm" datetime format.
+     *
+     * @param input The date string to be checked.
+     * @return true if the input matches the datetime format, false otherwise.
+     */
+    private static boolean isDateTimeFormat(String input) {
+        try {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
+            LocalDateTime.parse(input, formatter);
+            return true;
+        } catch (DateTimeParseException e) {
+            return false;
+        }
+    }
+
+    /**
+     * Checks if the input string is in the "yyyy-MM-dd" date format.
+     *
+     * @param input The date string to be checked.
+     * @return true if the input matches the date format, false otherwise.
+     */
+    private static boolean isDateFormat(String input) {
```

Comment:
I like how you name your boolean fields and method. Very easy to tell that it is boolean.